ext/TensorKitChainRulesCoreExt.jl

```diff
@@ -0,0 +1,337 @@
+module TensorKitChainRulesCoreExt
+
+using TensorOperations
+using TensorKit
+using ChainRulesCore
+using LinearAlgebra
+using TupleTools
+
+# Utility
+# -------
+
+_conj(conjA::Symbol) = conjA == :C ? :N : :C
+trivtuple(N) = ntuple(identity, N)
+
+function _repartition(p::IndexTuple, N₁::Int)
+    length(p) >= N₁ ||
+        throw(ArgumentError("cannot repartition $(typeof(p)) to $N₁, $(length(p) - N₁)"))
+    return p[1:N₁], p[(N₁ + 1):end]
+end
+_repartition(p::Index2Tuple, N₁::Int) = _repartition(linearize(p), N₁)
+function _repartition(p::Union{IndexTuple,Index2Tuple}, ::Index2Tuple{N₁}) where {N₁}
+    return _repartition(p, N₁)
+end
+function _repartition(p::Union{IndexTuple,Index2Tuple},
+                      ::AbstractTensorMap{<:Any,N₁}) where {N₁}
+    return _repartition(p, N₁)
+end
+
+TensorKit.block(t::ZeroTangent, c::Sector) = t
```

Comment:
This seems a bit suspicious? Why is this needed?

Reply:
This is mostly to avoid having to manually deal with the `ZeroTangent` type. For example, a QR decomposition where the final result does not depend on R, would generate a `ZeroTangent` for dR, which is just an abstract representation that behaves as the zero vector in any (co)vectorspace. As some of the `rrules` are implemented "blockwise", this would either require manually checking if a tangent is a `ZeroTangent`, or, which is what I chose to do, rely on the hope that the compiler would recognize that the blockwise operation results in ZeroTangents anyways, and thus automatically takes care of this